src/Git/Value/SemVerVersion.php

```diff
@@ -86,6 +86,19 @@ public function isNewMajorRelease(): bool
 
     public function lessThanEqual(self $other): bool
     {
-        return $this->fullReleaseName() <= $other->fullReleaseName();
+        return $this->compare($other) <= 0;
+    }
+
+    private function compare(self $other): int
+    {
+        $comparison = $this->major <=> $other->major;
+
+        if ($comparison !== 0) {
+            return $comparison;
+        }
+
+        $comparison = $this->minor <=> $other->minor;
+
+        return $comparison !== 0 ? $comparison : $this->patch <=> $other->patch;
```

Comment:
this can all be condensed to:
```php
    return [$this->major, $this->minor, $this->patch] <=> [$other->major, $other->minor, $other->patch];
```

Reply:
@bendavies cool, didn't know. Would you like to send a patch?

Reply:
sure!

Reply:
Why not use something like:
- https://github.com/composer/semver

it's been battle-tested since 2013 or so being part of the composer package.

Reply:
Could be done, but this code was supposed to work on a very isolated scenario, so IMO, since it is tested and relatively well protected from mutants, introducing a dependency here will likely raise more issues due to the use-case scenario being widened (while it is intentionally narrow)